src/test/unittest/basetest.py

```diff
@@ -208,15 +208,18 @@ def _execute(self):
             config_str = '{}/{}/{}'.format(self.test_type, c.fs, c.build)
             if self.valgrind:
                 config_str = '{}/{}'.format(config_str, self.valgrind)
-            self.msg.print('{}: SETUP\t({})'.format(self, config_str))
 
-            # removes old log files, to make sure that logs made by test
-            # are up to date
-            self.remove_log_files()
-
-            self.setup(c)
-            start_time = datetime.now()
             try:
+                if self.valgrind and self.valgrind.valgrind_exe is None:
```

Comment:
This should be printed out in valgrind.py.
Can we avoid having any dependency on "self.valgrind" variable in this file?